evalml/automl/automl_algorithm/iterative_algorithm.py

```diff
@@ -124,7 +124,10 @@ def _transform_parameters(self, pipeline_class, proposed_parameters):
         parameters = {}
         if self._pipeline_params:
             parameters['pipeline'] = self._pipeline_params
-        component_graph = [handle_component_class(c) for c in pipeline_class.component_graph]
+        if isinstance(pipeline_class.component_graph, list):
```

Comment:
I think this logic would be easier to maintain if this lived in the `PipelineBase` implementation. This can be a `linearized_component_graph` classproperty?

Reply:
I'm not quite sure what you mean - would `linearized_component_graph` be a boolean flag indicating if the pipeline is linear, or a list of the components, regardless of if the graph is a dictionary or list?

Reply:
If you're thinking a list of the components in a linear order, would it make sense to combine this with [your other suggestion](https://github.com/alteryx/evalml/pull/1543#discussion_r543584707) above? Some sort of property that contains the component instances in a list

Reply:
I was thinking of the list of component classes, regardless if the graph is a dictionary or a list!

Reply:
I think it would be tough to combine it with the other suggestion because here we don't have the pipeline instance, just the class definition so I don't think you'll be able to grab the component instances. That being said, if there's a way to combine them that I'm not seeing that would be great!

Also, `linearized_component_graph` is just the first thing that came to mind 😂  I'm sure there are a billion better names for it hehe.

Reply:
Gotcha - so this class property wouldn't necessarily guarantee computation order then, just some linear order of components that may or may not make sense

Reply:
Yea in this case the order would not matter!